test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -0,0 +1,409 @@
+    }
+    private val followingEdgeInsets: EdgeInsets by lazy {
+        EdgeInsets(
+            mapboxMap.getSize().height.toDouble() * 2.0 / 3.0,
+            0.0 * pixelDensity,
+            0.0 * pixelDensity,
+            0.0 * pixelDensity
+        )
+    }
+    private val currentManeuverCallback = object : ManeuverCallback {
+        override fun onManeuver(currentManeuver: ManeuverState.CurrentManeuver) {
+            val maneuver = currentManeuver.maneuver
+            val primary = maneuver.primary
+            val secondary = maneuver.secondary
+            val sub = maneuver.sub
+            val lane = maneuver.laneGuidance
+            if (secondary?.componentList != null) {
+                maneuverView.render(ManeuverState.ManeuverSecondary.Show)
+                maneuverView.render(ManeuverState.ManeuverSecondary.Instruction(secondary))
+            } else {
+                maneuverView.render(ManeuverState.ManeuverSecondary.Hide)
+            }
+            if (sub?.componentList != null) {
+                maneuverView.render(ManeuverState.ManeuverSub.Show)
+                maneuverView.render(ManeuverState.ManeuverSub.Instruction(sub))
+            } else {
+                maneuverView.render(ManeuverState.ManeuverSub.Hide)
+            }
+            maneuverView.render(ManeuverState.ManeuverPrimary.Instruction(primary))
+            maneuverView.render(ManeuverState.UpcomingManeuvers.RemoveUpcoming(maneuver))
+            if (lane != null) {
+                maneuverView.render(ManeuverState.LaneGuidanceManeuver.Show)
+                maneuverView.render(ManeuverState.LaneGuidanceManeuver.AddLanes(lane))
+            } else {
+                maneuverView.render(ManeuverState.LaneGuidanceManeuver.Hide)
+                maneuverView.render(ManeuverState.LaneGuidanceManeuver.RemoveLanes)
+            }
+        }
+    }
+
+    private val stepDistanceRemainingCallback = object : StepDistanceRemainingCallback {
+        override fun onStepDistanceRemaining(
+            distanceRemaining: ManeuverState.DistanceRemainingToFinishStep
+        ) {
+            maneuverView.render(distanceRemaining)
+        }
+    }
+
+    private val upcomingManeuversCallback = object : UpcomingManeuversCallback {
+        override fun onUpcomingManeuvers(state: ManeuverState.UpcomingManeuvers.Upcoming) {
+            maneuverView.render(state)
+        }
+    }
+
+    private val locationObserver = object : LocationObserver {
+        override fun onRawLocationChanged(rawLocation: Location) { }
+        override fun onEnhancedLocationChanged(
+            enhancedLocation: Location,
+            keyPoints: List<Location>
+        ) {
+            navigationLocationProvider.changePosition(
+                enhancedLocation,
+                keyPoints,
+            )
+            viewportDataSource.onLocationChanged(enhancedLocation)
+            viewportDataSource.evaluate()
+            navigationCamera.requestNavigationCameraToFollowing()
```

Comment:
We are requesting the following here on each update, this should only happen once.

Reply:
If I don't do this the camera doesn't enter the following mode.

Reply:
I believe it does, it happens on the start navigation button click.